packages/x-date-pickers/src/locales/ukUA.ts

```diff
@@ -0,0 +1,73 @@
+import { PickersLocaleText } from './utils/pickersLocaleTextApi';
+import { getPickersLocalization } from './utils/getPickersLocalization';
+import { CalendarPickerView } from '../internals/models';
+
+// This object is not Partial<PickersLocaleText> because it is the default values
+
+const ukUAPickers: Partial<PickersLocaleText<any>> = {
+  // Calendar navigation
+  previousMonth: 'Попередній місяць',
+  nextMonth: 'Наступний місяць',
+
+  // View navigation
+  openPreviousView: 'відкрити попередній вигляд',
+  openNextView: 'відкрити наступний вигляд',
+  calendarViewSwitchingButtonAriaLabel: (view: CalendarPickerView) =>
+    view === 'year'
+      ? 'річний вигляд відкрито, перейти до календарного вигляду'
+      : 'календарний вигляд відкрито, перейти до річного вигляду',
+  inputModeToggleButtonAriaLabel: (isKeyboardInputOpen: boolean, viewType: 'calendar' | 'clock') =>
+    isKeyboardInputOpen
+      ? `текстове поле відкрите, перейти до  ${viewType} вигляду`
+      : `${viewType} вигляд наразі відкрито, перейти до текстового поля`,
+
+  // DateRange placeholders
+  start: 'Початок',
+  end: 'Кінець',
+
+  // Action bar
+  cancelButtonLabel: 'Відміна',
+  clearButtonLabel: 'Очистити',
+  okButtonLabel: 'OK',
+  todayButtonLabel: 'Сьогодні',
+
+  // Toolbar titles
+  datePickerToolbarTitle: 'Вибрати дату',
+  dateTimePickerToolbarTitle: 'Вибрати дату і час',
+  timePickerToolbarTitle: 'Вибрати час',
+  dateRangePickerToolbarTitle: 'Вибрати календарний період',
+
+  // Clock labels
+  clockLabelText: (view, time, adapter) =>
+    `Select ${view}. ${
+      time === null ? 'Час не вибраний' : `Вибрано час ${adapter.format(time, 'fullTime')}`
+    }`,
+  hoursClockNumberText: (hours) => `${hours} годин`,
+  minutesClockNumberText: (minutes) => `${minutes} хвилин`,
```

Comment:
As potential improvement, we can use different plural forms depending on the number like we do in the data grid:
https://github.com/mui/mui-x/blob/099729f6704570575b0a395a6efac471449d0888/packages/grid/x-data-grid/src/locales/ukUA.ts#L46-L51

Reply:
Sorry, I saw your profile picture, then I thought you did a language review but it was your self-assignment 🙈

Reply:
@alexfauquette no worries 🙂
@Dufran Feel free to open a follow-up PR if you want